src/app/model/ChessEngine.ts

```diff
@@ -1,3 +1,37 @@
 import { ChessModel } from './ChessModel';
+import { PieceMoves } from './PieceMoves';
+import { PiecePositions } from './PiecesPositions';
+import { PieceWasCaptured, PieceWasMoved, Square } from './Types';
+import { Piece } from './Piece';
+import { Player } from './Player';
+import { ChessBoard } from './ChessBoard';
 
-export class ChessEngine implements ChessModel {}
+export class ChessEngine implements ChessModel, PieceMoves, PiecePositions {
+  move(byPlayer: Player, squareFrom: Square, squareTo: Square, board: ChessBoard): (PieceWasMoved | PieceWasCaptured)[] {
```

Comment:
Potem jak zrobisz ta metode, to dodaj jej deklaracje do ChessModel. Bo do ChessModel bedziemy sie odnosic z prezentera. Jego implementacja bedzie wlasnie ten ChessEngine :)